warp10/src/main/java/io/warp10/script/WarpScriptStack.java

```diff
@@ -428,6 +428,20 @@ public String snapshot(boolean hideSecure) {
             if (o instanceof Macro) {
               sb.append(((Macro) o).snapshot(hideSecure));
               sb.append(" ");
+            } else if (o instanceof WarpScriptStackFunction) {
+              String funcSnapshot = o.toString();
+              // In the case the snapshot of the function is 'MYFUNC' FUNCREF, instead of adding
+              // 'MYFUNC' FUNCREF EVAL to the snapshot, MYFUNC can simply be added.
+              if (o instanceof NamedWarpScriptFunction
+                  && null != funcSnapshot
+                  && funcSnapshot.equals("'" + ((NamedWarpScriptFunction) o).getName() + "' " + WarpScriptLib.FUNCREF)) {
```

Comment:
This construct should be delegated to a static function of NamedWarpScriptFunction to avoid having to track where it is used throughout the code base should it change.